Pod/Classes/OTNoop.m

```diff
@@ -24,7 +24,9 @@ @implementation OTNoopSpan
 - (id<OTTracer>)tracer { return g_defaultNoopTracer; }
 - (id<OTSpanContext>)context { return g_defaultNoopSpanContext; }
 - (void)setOperationName:(NSString*)operationName {}
-- (void)setTag:(NSString*)key value:(NSString*)value {}
+- (void)setTag:(NSString*)key stringValue:(nonnull NSString *)value {}
```

Comment:
`NSObject` covers everything that's an `NSObject` subclass - which includes all object types. So if we are passing in things like `NSString`/`NSNumber`/`NSDictionary`/`NSArray`, it'd be no problem. We have `NSString` for **strings** and `NSNumber` for **numerics**, both of which can be covered by `NSObject`. However here we need to cover tag value being a **boolean**, which is a primitive and not an object. 

Method overloading is not supported in ObjC, so we _can't_ have both of these at the same time:
`- (void)setTag:(NSString *)key value:(NSObject *)value;`
`- (void)setTag:(NSString *)key value:(BOOL)value;`

One option is to have 2 methods: 
1. `- (void)setTag:(NSString *)key value:(NSObject *)value;` to cover strings (NSString) and numerics (NSNumber) 
2. `- (void)setTag:(NSString *)key boolValue:(BOOL)value;` to cover booleans (BOOL). 

This seems more... lopsided.. 

By having 3 methods each covering a specific type of value they support, we make it explicit what value type each one support. This also prevents callers from passing in value types not supported by the spec, like dictionaries and arrays.

I guess one example is the `NSNumber` class where you can construct an `NSNumber` from various primitive types, and the initializers are explicitly named for each type: https://developer.apple.com/reference/foundation/nsnumber?language=objc